src/main/java/org/jitsi/videobridge/cc/SimulcastController.java

```diff
@@ -722,6 +722,37 @@ void setTL0Idx(int newTL0Idx)
         }
 
         /**
+         * Given a list of available quality indices and a target index, find
+         * the highest available quality index <= targetIdx
+         * @param availableQualityIndices the available quality indices
+         * @param targetIdx the target index
+         * @return the highest quality index <= targetIdx
+         * //FIXME(brian): this could be static, but can't add a static method
+         * to the inner class.  should it be put elsewhere and made a static
+         * helper method? (the outer class?)
+         */
+        private int findMaximumQualityIndex(int[] availableQualityIndices, int targetIdx)
+        {
+            int maxIndex = -1;
+            if (availableQualityIndices != null)
+            {
+                for (int i = 0; i < availableQualityIndices.length; ++i)
```

Comment:
i don't quite follow your specific example (maybe there's a typo?) but i think i see what you're referring to...you're right, but i'm confused by this actually.  the old logic effectively means we could choose a quality index that was larger than than our target which doesn't seem correct? e.g.

```
availableQualityIndices = {10, 11, 12, 13}
targetIdx = 4;
```

we would do `currentIdx = availableQualityIndices[0];` without checking if it was `<= targetIdx` before, right?